app/objects/c_report.py

```diff
@@ -60,3 +60,142 @@ def store(self, ram):
         existing.update('hosts', self.hosts)
         existing.update('network_map', self.network_map)
         return existing
+
+    def retrieve_host_by_id(self, id):
+        return self.hosts[id]
+
+    # 
+    # Getters / Setters
+    # 
+
+    def get_host_freebie_abilities(self, id):
+        if id in self.hosts:
+            return self.hosts[id].freebie_abilities
+        return None
+
+    def set_host_freebie_abilities(self, id, abilities):
+        if id in self.hosts:
+            self.hosts[id].freebie_abilities = abilities
+            return True
+        return False
+
+    def get_host_acces(self, id):
+        if id in self.hosts:
+            return self.hosts[id].access
+        return None
+    
+    def set_host_access(self, id, access):
+        if id in self.hosts:
+            self.hosts[id].access = access
+            return True
+        return False
+
+    def get_host_denied_abilities(self, id):
+        if id in self.hosts:
+            return self.hosts[id].denied_abilities
+        return None
+
+    def set_host_denied_abilities(self, id, denied):
+        if id in self.hosts:
+            self.hosts[id].denied_abilities = denied
+            return True
+        return False
+
+    def get_host_ip(self, id):
+        if id in self.hosts:
+            return self.hosts[id].ip
+        return None
+
+    def set_host_ip(self, id, ip):
+        if id in self.hosts:
+            self.hosts[id].ip = ip
+            return True
+        return False
+
+    def get_host_ports(self, id):
+        if id in self.hosts:
+            return self.hosts[id].ports
+        return None
+
+    def set_host_ports(self, id, ports):
```

Comment:
For nested objects, we might want to consider specifying the expected types, (here a list of second class `Port` objects). This comment applies to some of the other custom objects like `OS` or `Software` as well.